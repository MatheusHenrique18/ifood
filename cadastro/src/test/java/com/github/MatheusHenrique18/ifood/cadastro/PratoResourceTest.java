package com.github.MatheusHenrique18.ifood.cadastro;

import static io.restassured.RestAssured.given;

import javax.ws.rs.core.Response.Status;

import org.approvaltests.JsonApprovals;
import org.junit.jupiter.api.Test;

import com.github.database.rider.cdi.api.DBRider;
import com.github.database.rider.core.api.configuration.DBUnit;
import com.github.database.rider.core.api.configuration.Orthography;
import com.github.database.rider.core.api.dataset.DataSet;

import io.quarkus.test.common.QuarkusTestResource;
import io.quarkus.test.junit.QuarkusTest;
import io.restassured.http.ContentType;

@DBRider
@DBUnit(caseInsensitiveStrategy = Orthography.LOWERCASE)
@QuarkusTest
@QuarkusTestResource(CadastroTestLifeCycleManager.class)
public class PratoResourceTest {

	@Test
    @DataSet("pratos-cenario-1.yml")
    public void testBuscarPratos() {
		Long parameterValue = 123L;
        String resultado = given()
					.contentType(ContentType.JSON)
					.with().pathParam("idRestaurante", parameterValue)
					.when().get("/restaurantes/{idRestaurante}/pratos")
					.then()
					.statusCode(Status.OK.getStatusCode())
					.extract().asString();
        JsonApprovals.verifyJson(resultado);
    }
	
}
