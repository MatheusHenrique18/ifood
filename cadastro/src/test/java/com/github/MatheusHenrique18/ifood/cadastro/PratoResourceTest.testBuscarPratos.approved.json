[
  {
    "descricao": "teste prato 1",
    "idPrato": 12345,
    "nome": "teste chocolate",
    "preco": 10.00,
    "restaurante": {
      "idRestaurante": 123,
      "proprietario": "ID do keycloack"
    }
  },
  {
    "descricao": "teste prato 2",
    "idPrato": 6789,
    "nome": "teste ovo",
    "preco": 12.00,
    "restaurante": {
      "idRestaurante": 123,
      "proprietario": "ID do keycloack"
    }
  }
]